### PR TITLE
Apollo Client 3 (@apollo/client) updates

### PR DESCRIPTION
**Pull request by hwillson**
This PR updates devtools to be compatible with both Apollo Client 2 and 3. The commit messages go into more details, but there is one important thing to call out. Since we're working on a new version of devtools that is essentially a rewrite of the existing codebase, I've done the minimum necessary to get the existing devtools working with AC3. In most cases this is okay, but it does mean there are slight differences/limitations in the "Cache" panel. From https://github.com/apollographql/apollo-client-devtools/commit/5b256d6a8b45390597b24c9a7ff2c9d23fac2602: 

> We're working on a complete re-write of the AC devtools, so this
> commit should be considered temporary.
> 
> Apollo Client's internal cache structure has changed between
> AC2 and AC3. This means the `Inspector` code bits of devtools
> (that drive the Cache panel) are no longer compatible with the
> newer cache structure. The current `Inspector` relies on AC2's
> internal cache structure to render a hierarchical view of the
> cache, with links between different entities. Re-creating this
> for AC3 will take time away from our newer devtools rewrite
> work, so this commit focuses on making the minimal changes
> needed to get the Cache panel working with AC3.
> 
> With this commit, AC3 cache contents are displayed in the
> Cache panel, but contents are not properly linked (e.g. cache
> references are not followed and linked together). Nested
> entities are displayed as simple JSON dumps.

In other words, this is how the "Cache" panel looks/behaves when used with AC2:

Image 2020-05-25 at 1 58 25 PM png

Whereas this is how it looks when used with AC3:

Image 2020-05-25 at 1 59 16 PM png

The AC3 version dumps a full representation of the cache contents, but it does not automatically link in referenced entities. The devtools rewrite we're working on will take care of all of this (and more), but hopefully this will serve as a stop gap until the new devtools are ready.

Fixes https://github.com/apollographql/apollo-client-devtools/issues/251.

### CODE REVIEW

**Comment by Leo4815162342** [comment]
@benjamn shall we merge this temp solution?
